js/modules/k6/http/request.go

```diff
@@ -310,6 +312,26 @@ func (h *HTTP) parseRequest(
 				case *HTTPCookieJar:
 					result.ActiveJar = v.jar
 				}
+			case "compression":
+				var algo = params.Get(k).ToString().String()
+				if algo != "gzip" {
+					return nil, fmt.Errorf("unknown compression algorithm %s, only supported algorithm is gzip", algo)
+				}
+
+				var buf bytes.Buffer
+				var w = gzip.NewWriter(&buf)
```

Comment:
`parseRequest()` in `js/modules/k6/http/request.go` doesn't seem like the right place to put this... Here we should just parse the user's intention to use `gzip` or `deflate` or whatever as `Content-Encoding`, and probably validate the parameters, but the actual implementation should be in the generic HTTP request making code in `lib`, I think...

And even if that wasn't the case due to code structure reasons, we still shouldn't add more code here when the method is over 250 lines of code and has `//TODO break this function up` and `//nolint: gocyclo` in front of it :smile:

Reply:
I would argue that if we do it with `Content-Encoding` - yes, otherwise I would argue that function needs to be broken down and other parts too need to move to the `httpext` package. 
On the other side I would argue that changing compression to be activated through setting  `Content-Encoding` in headers will mean that 
1. We will need to either not throw errors for unknown Content-Encodings leaving users to wonder why `gizp` doesn't work. Or not let them set Content-Encoding and encode their own bodies
2. Not let people gzip/deflate their own bodies - which seems rare
3. Have "magical" behaviour just by setting a header - I dislike this much the most

cc @robingustafsson

Reply:
As I agreed in the PR comments, I'm fine with sticking with `compression` in the `http.Params` object.

But I would still argue that this isn't the place for this code. Here, we should just set a property of the `ParsedHTTPRequest` struct, and we should actually handle the compression of the body in the `lib/httpext` code. Here are some very concrete reasons why:
- it's not clear whether `case "compression":` or `case "headers":` will be reached first... 
- here we're just assembling and validating the intentions of the user into a unified typed form (i.e. the `ParsedHTTPRequest`), it's not the place for "business" logic
- if we ever want to split this function up, it's not clear that the code that parses the params should actually have access to the body...